### PR TITLE
#1348 Release Trigger Action Bug fix

### PR DESCRIPTION
**Pull request by ldraney**
# Description

Addresses #1348
The Release Trigger action fails to initiate a deploy to Prod when a Release is Published.
This PR is move what was once logical if statements into two trigger files.
image
- This also adds a syntax improvement to improve reliability. 
- release.yml just needed to be renamed (should have been renamed earlier), but it's structure doesn't need to change from what is in master. 

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- [x] The POC video can be [found here](https://drive.google.com/file/d/1jGHNkhmAUSBMBNJY63hDEu7Gtzr8-JDA/view?usp=sharing)
- [x] The document I wrote out about why separating triggers to different files can be [found here](https://docs.google.com/document/d/1GjB-77RT1odUZC8TumpTK0pLKJJ7TxHZJh0vIb3BpFY/edit)
- [x] Test Repo code release can be [found here](https://github.com/ldraney/dotfilesWSL/tree/separate3/.github/workflows), as well as the triggered runs:
image

## Checklist
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/pre-release_trigger.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    secrets: inherit
+
+  release:
+    needs: [setup-release]
+    if: ${{ github.event.release.prerelease == false }}
+    uses: ./.github/workflows/release.yml
+    with:
+      environment: "prod"
+      git-tag: "${{ github.event.release.tag_name }}"
+    secrets: inherit
```
Comment (cris-oddball) [inline .github/workflows/pre-release_trigger.yml:40]: why is this still needed here? this workflow should only be for the pre-release.

File: .github/workflows/release_trigger.yml [hunk 2]
```diff
@@ -19,11 +19,11 @@ jobs:
           echo "Release tag: ${{ github.event.release.tag_name }}"	          
           echo "Release name: ${{ github.event.release.name }}"
           echo "is a prerelease? (github.event.release.prerelease): ${{ github.event.release.prerelease }}"
-          echo "is a regular release? (will say release if true)(github.event.action): ${{ github.event.action }}"
+          echo "is published (staging) or a release (prod)?: ${{ github.event.action }}"
```
Comment (cris-oddball) [inline .github/workflows/release_trigger.yml:22]: Why is staging in here?

File: .github/workflows/release_trigger.yml [hunk 3]
```diff
@@ -19,11 +19,11 @@ jobs:
...
 
   pre-release:
     needs: [setup-release]
-    if: github.event.release.prerelease == true
+    if: ${{ github.event.release.prerelease == true }}
     uses: ./.github/workflows/release.yml
     with:
       environment: "staging"
```
Comment (cris-oddball) [inline .github/workflows/release_trigger.yml:29]: why is a pre-release needed here?

**Review by cris-oddball — CHANGES_REQUESTED** [review]
please rename `.github/workflows/pre-release_trigger.yml` to `.github/workflows/prerelease_trigger.yml` without mixing dashes and underscores.

File: .github/workflows/release_trigger.yml (hunk 2 above)
Comment (ldraney) [reply .github/workflows/release_trigger.yml:22]: Simply as a string check to be sure

File: .github/workflows/release_trigger.yml (hunk 3 above)
Comment (ldraney) [reply .github/workflows/release_trigger.yml]: removed

**Review by cris-oddball — DISMISSED** [review]

**Review by cris-oddball — APPROVED** [review]

**Review by kalbfled — APPROVED** [review]

**Pull request merged**